src/api/chats.js

```diff
@@ -14,11 +14,17 @@ const socketHelpers = require('../socket.io/helpers');
 
 const chatsAPI = module.exports;
 
-function rateLimitExceeded(caller) {
+async function rateLimitExceeded(caller) {
 	const session = caller.request ? caller.request.session : caller.session; // socket vs req
 	const now = Date.now();
+	const [isAdmin, reputation] = await Promise.all([
+		user.isAdministrator(caller.uid),
```

Comment:
Maybe change this to user.isPrivileged to let global mods and moderators through even if they have 0 rep.